float32 direction
float32 distance

# 0 : T-junction
# 1 : X-junction
# 2 : left door
# 3 : right door

int32 goal_type
---
---
float32 distance_travelled
float32 current_direction
float32 desired_direction
bool recovery_mode